sale_order_line_packaging_qty/models/sale_order_line.py

```diff
@@ -0,0 +1,101 @@
+        for sol in self:
+            if (
+                not sol.product_packaging
+                or sol.product_uom_qty == 0
+                or sol.product_packaging.qty == 0
+            ):
+                sol.product_packaging_qty = 0
+                continue
+            # Consider uom
+            if sol.product_id.uom_id != sol.product_uom:
+                product_qty = sol.product_uom._compute_quantity(
+                    sol.product_uom_qty, sol.product_id.uom_id
+                )
+            else:
+                product_qty = sol.product_uom_qty
+            sol.product_packaging_qty = product_qty / sol.product_packaging.qty
+
+    def _inverse_product_packaging_qty(self):
+        for sol in self:
+            if not sol.product_packaging:
+                raise UserError(
+                    _(
+                        "You must define a package before setting a quantity "
+                        "of said package."
+                    )
+                )
+            if sol.product_packaging.qty == 0:
+                raise UserError(
+                    _("Please select a packaging with a quantity bigger than 0")
+                )
+            sol.write(
+                {
+                    "product_uom_qty": sol.product_packaging.qty
+                    * sol.product_packaging_qty,
+                    "product_uom": sol.product_packaging.product_uom_id.id,
+                }
+            )
+
+    @api.onchange("product_packaging")
+    def _onchange_product_packaging(self):
+        if self.product_packaging:
+            self.update(
+                {
+                    "product_packaging_qty": 1,
+                    "product_uom_qty": self.product_packaging.qty,
+                    "product_uom": self.product_id.uom_id,
+                }
+            )
+        else:
+            self.update({"product_packaging_qty": 0})
+        return super()._onchange_product_packaging()
+
+    @api.onchange("product_packaging_qty")
+    def _onchange_product_packaging_qty(self):
+        if self.product_packaging_qty and not self.product_packaging:
+            raise UserError(
+                _(
+                    "You must define a package before setting a quantity "
+                    "of said package."
+                )
+            )
+        else:
+            self.update(
+                {
+                    "product_uom_qty": self.product_packaging_qty
+                    * self.product_packaging.qty,
+                    "product_uom": self.product_id.uom_id,
+                }
+            )
+
+    @api.onchange("product_uom_qty")
+    def _onchange_product_uom_qty(self):
+        """
+        Ensure a warning is raised when changing the package if the qty
+        is not a multiple of the package qty.
+        """
+        # TODO Drop once https://github.com/odoo/odoo/pull/49150/ is merged
+        res = super()._onchange_product_uom_qty()
+        if not res:
```

Comment:
```suggestion
        if not res and not (self.state == 'sale' and self.product_id.type in ['product', 'consu'] and self.product_uom_qty < product_uom_qty_origin and self.product_uom_qty < self.qty_delivered):
```
according to your PR to odoo

Reply:
I'm not sure to understand why we'd need this condition as we only want to ensure `_check_package` is called when `product_uom_qty` is changed :no_mouth: